opm/parser/eclipse/EclipseState/Schedule/MSW/Segment.hpp

```diff
@@ -67,6 +69,10 @@ namespace Opm {
 
         const std::shared_ptr<SpiralICD>& spiralICD() const;
 
+        void updateValve(const Valve& valve, const double segment_length);
+
+        const std::shared_ptr<Valve>& valve() const;
```

Comment:
Note that this will return (a reference to) a constant `shared_ptr` to a mutable `Valve`, meaning the compiler will not prevent anyone from misusing this as
```C++
segment.valve()->setPipeDiameter(1.0e+5);
```
Is that really intentional?

Reply:
It is not intentional and should be fixed.

Reply:
> It is not intentional and should be fixed.

Okay, may I then suggest we return a raw pointer instead?  Something along the lines of
```C++
const Valve* valve() const
{
    return m_valve.get();
}
```

Reply:
sure. I can do that.